### ISSUE TITLE
Various invite file parsing issues

### ISSUE THREAD

**Issue by numinit** [gsliepen/tinc#165]
tinc version: `tinc version 1.1pre15 (built Dec 26 2017 21:04:50, protocol 17.7)`

The following `invitation-created` script produces a nonsensical warning when the invitation is accepted, seemingly due to the single empty line.

```sh
#!/bin/sh

cat > "$INVITATION_FILE" <<EOF
Name = $NODE
Mode = switch
Netname = $NETNAME
ConnectTo = $NAME
Ifconfig = dhcp

# [Entry node configuration]
EOF

tinc export >> "$INVITATION_FILE"
```

```
$ tinc join host/key
Connected to host port 655...
Ignoring unknown variable '' in invitation.
...........................................................+++ p
...............................................+++ q

Please review the following tinc-up script:

#!/bin/sh
dhclient -nw "$INTERFACE"
ip link set "$INTERFACE" up

Do you want to use this script [y]es/[n]o/[e]dit? y

tinc-up enabled.
Configuration stored in: /etc/tinc/netname
Invitation succesfully accepted.
```

**Comment by numinit** [comment]
Here's an otherwise identical script that makes the client fail to connect:

```sh
#!/bin/sh

cat > "$INVITATION_FILE" <<EOF
# [Client configuration]
Name = $NODE
Mode = switch
Netname = $NETNAME
ConnectTo = $NAME
Ifconfig = dhcp

# [Entry node configuration]
EOF

tinc export >> "$INVITATION_FILE"
```

**Comment by gsliepen** [comment]
I just fixed the issue with the warnings for empty lines. The problem with comments at the start of the invitation file are the result of tinc expecting the first line to always contain `Name = ...`. It would make parsing invitation files harder to allow comments and/or empty lines at the start, so for now I just added to the documentation that the first line of the invitation file must always be `Name = ...`.
